driver-core/src/main/com/mongodb/event/EventListenerSettings.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.event;
+
+import com.mongodb.annotations.NotThreadSafe;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.mongodb.assertions.Assertions.notNull;
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * Settings for the various event listeners.
+ *
+ * @since 3.5
+ */
+public final class EventListenerSettings {
```

Comment:
The reasoning is to simplify the configuring the of event listeners for users.  Currently for Async users they are scattered around making it difficult to configure - I don't think they should be penalised in the API.

Had we opted for a specific settings originally then adding new listeners would have been much easier and kept the API fit for purpose.  The `ClusterFactory` interface is the case in point, the interface is now not fit for purpose and because we can't change it, the `DefaultClusterFactory` has grown over time to handle new settings.

As this pattern is integral I'll pause any further work until we agree on the pattern to use.